### ISSUE TITLE
Possible false-positive: Cannot instantiate protocol class

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#1956]
Starting with v1.1.147 pyright is flagging the following code:

```python
from typing import Protocol, TypeVar, Type

T1 = TypeVar("T1", covariant=True)
T2 = TypeVar("T2")


class A(Protocol[T1]):
    def __init__(self) -> None:
        ...

def f(x: T2) -> Type[A[T2]]:
    ...

a_class = f(1)
a_instance: A[int] = reveal_type(a_class())  # error: Cannot instantiate protocol class "A" (reportGeneralTypeIssues)
```

**Expected behavior**
I would not expect `-> Type[A[int]]` to indicate that the actual protocol-class itself is being returned, but rather that a class object that satisfies the protocol `Type[A]` is being returned. 

If this is not correct, can you recommend a proper way for me to handle this sort of annotation?

**Comment by erictraut** [comment]
Thanks for reporting the bug.

This is a regression I introduced when implementing this [feature request](https://github.com/microsoft/pyright/issues/1936). I didn't take into consideration this case.

**Comment by erictraut** [comment]
This will be fixed in the next release. Apologies for any inconvenience the regression caused.

**Comment by rsokl** [comment]
No problem. Thank you for the quick response and for the absolutely fantastic work on pyright.

**Comment by erictraut** [comment]
This is included in pyright 1.1.148, which I just published. It will also be included in the next release of pylance.
